### PR TITLE
RDKB-50202:mqtt_webcfg RFC Enable/Disable to be control dynamically

### PR DESCRIPTION
**Pull request by nlrcomcast**

### CODE REVIEW

File: src/webcfg_mqtt.c [hunk 1]
```diff
@@ -160,7 +163,32 @@ void* WebconfigMqttTask(void *status)
...
 		mqttSubscribeInit();
 	}
+	pthread_cond_wait(&mqtt_sync_condition, &mqtt_sync_mutex);
+	webcfg_onconnect_flag = 0;
+	WebcfgInfo("rbus event Unsubscribe to mqtt subscribed callback\n");
+	rc = rbusEvent_Unsubscribe(rbus_handle, WEBCFG_SUBSCRIBE_CALLBACK);
+	if(rc != RBUS_ERROR_SUCCESS)
+	{
+		WebcfgError("consumer: rbusEvent_Unsubscribe for %s failed: %d\n", WEBCFG_SUBSCRIBE_CALLBACK, rc);
+		return NULL;
```
Comment (harnishypatel) [inline src/webcfg_mqtt.c:173]: pthread_mutex_unlock (&mqtt_sync_mutex); missing ?

File: src/webcfg_mqtt.c [hunk 2]
```diff
@@ -160,7 +163,32 @@ void* WebconfigMqttTask(void *status)
...
+	}
+
+	WebcfgInfo("rbus event Unsubscribe to mqtt message callback\n");
+	rc = rbusEvent_Unsubscribe(rbus_handle, WEBCFG_ONMESSAGE_CALLBACK);
+	if(rc != RBUS_ERROR_SUCCESS)
+	{
+		WebcfgError("consumer: rbusEvent_Unsubscribe for %s failed: %d\n", WEBCFG_ONMESSAGE_CALLBACK, rc);
```
Comment (harnishypatel) [inline src/webcfg_mqtt.c:180]: pthread_mutex_unlock (&mqtt_sync_mutex); missing ?

File: src/webcfg_mqtt.c [hunk 3]
```diff
@@ -160,7 +163,32 @@ void* WebconfigMqttTask(void *status)
...
+		return NULL;
+	}
 
+	WebcfgInfo("rbus event Unsubscribe to mqtt publish callback\n");
+	rc = rbusEvent_Unsubscribe(rbus_handle, WEBCFG_ONPUBLISH_CALLBACK);
+	if(rc != RBUS_ERROR_SUCCESS)
+	{
+		WebcfgError("consumer: rbusEvent_Unsubscribe for %s failed: %d\n", WEBCFG_ONPUBLISH_CALLBACK, rc);
```
Comment (harnishypatel) [inline src/webcfg_mqtt.c:188]: pthread_mutex_unlock (&mqtt_sync_mutex); missing ?

File: src/webcfg_mqtt.c (hunk 1 above)
Comment (nlrcomcast) [reply src/webcfg_mqtt.c]: removed return, so pthread_mutex_unlock not adding

File: src/webcfg_mqtt.c (hunk 2 above)
Comment (nlrcomcast) [reply src/webcfg_mqtt.c:181]: removed return, so pthread_mutex_unlock not adding

File: src/webcfg_mqtt.c (hunk 3 above)
Comment (nlrcomcast) [reply src/webcfg_mqtt.c:188]: removed return, so pthread_mutex_unlock not adding

**Review by harnishypatel — APPROVED** [review]

**Pull request merged**